### PR TITLE
JDK-8324598: use mem_unit when working with sysinfo memory and swap related information

### PR DESCRIPTION
**Pull request by MBaesken**
According to the sysinfo manpage ( https://man7.org/linux/man-pages/man2/sysinfo.2.html ) the memory and swap related entries in the struct sysinfo are given as multiples of mem_unit bytes.
"In the above structure, sizes of the memory and swap fields are given as multiples of mem_unit bytes."

This is considered in most parts of the OpenJDK codebase when sysinfo is used but not all; should be added where it is missing.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8324598](https://bugs.openjdk.org/browse/JDK-8324598): use mem_unit when working with sysinfo memory and swap related information (**Bug** - P4)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [c46b804f](https://git.openjdk.org/jdk/pull/17550/files/c46b804f92ceb0a25812632e80004846955b6b81)
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17550/head:pull/17550` \
`$ git checkout pull/17550`

Update a local copy of the PR: \
`$ git checkout pull/17550` \
`$ git pull https://git.openjdk.org/jdk.git pull/17550/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17550`

View PR using the GUI difftool: \
`$ git pr show -t 17550`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17550.diff">https://git.openjdk.org/jdk/pull/17550.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17550#issuecomment-1907817988)

### CODE REVIEW

File: src/hotspot/os/linux/os_linux.cpp [hunk 1]
```diff
@@ -417,7 +417,7 @@ pid_t os::Linux::gettid() {
 julong os::Linux::host_swap() {
   struct sysinfo si;
   sysinfo(&si);
-  return (julong)si.totalswap;
+  return (julong)si.totalswap * si.mem_unit;
```
Comment (dholmes-ora) [inline src/hotspot/os/linux/os_linux.cpp:420]: Suggestion: `(julong)(si.totalswap * si.mem_unit);

File: src/java.base/linux/native/libjava/CgroupMetrics.c [hunk 2]
```diff
@@ -54,5 +54,5 @@ Java_jdk_internal_platform_CgroupMetrics_getTotalSwapSize0
     if (retval < 0) {
          return 0; // syinfo failed, treat as no swap
     }
-    return (jlong)si.totalswap;
+    return (jlong)si.totalswap * si.mem_unit;
```
Comment (dholmes-ora) [inline src/java.base/linux/native/libjava/CgroupMetrics.c:57]: Suggestion: `(julong)(si.totalswap * si.mem_unit);

**Review by dholmes-ora — APPROVED** [review]
Looks fine.

Minor nit with lack of parentheses.

Thanks

**Comment by MBaesken** [comment]
Thanks for the review! I followed your suggestions regarding parenthesis.

**Review by TheRealMDoerr — APPROVED** [review]
I would have preferred the original version, but it's ok either way.

**Comment by MBaesken** [comment]
Hi Martin, thanks for the review !

/integrate

**Pull request closed without merging**